suite/tests/client-interface/stolen-reg.dll.c

```diff
@@ -67,6 +67,59 @@ do_flush(app_pc next_pc)
     DR_ASSERT(false);
 }
 
+// Storing the original stolen reg before mconext set.
+static ptr_int_t orig_value = 0;
+
+// Random value to detect after mcontext set.
+static ptr_int_t test_value = 7;
+
+static void
+load_stolen_reg_to_mcontext()
+{
+    fprintf(stderr, "load_stolen_reg_to_mcontext entered\n");
+    void *drcontext = dr_get_current_drcontext();
+
+    fprintf(stderr, "test value = " IF_ARM_ELSE("%d", "%ld") "\n", test_value);
+
+    dr_mcontext_t mc;
+    mc.size = sizeof(mc);
+    mc.flags = DR_MC_ALL;
+
+    fprintf(stderr, "fetching TLS\n");
+
+    dr_get_mcontext(drcontext, &mc);
+
+    fprintf(stderr, "mc->stolen_reg after = " IF_ARM_ELSE("%d", "%ld") "\n",
+            mc.IF_ARM_ELSE(r10, r28));
```

Comment:
I think it is better to have a DR_ASSERT or otherwise explicit check here that r10==test_value.  That makes the code easier to read (it's clear what we expect) and is generally better than an implied, hard-to-figure-out assertion through the "golden output" check of the output file.

Reply:
Added the assert.  Just left the golden output though to expedite the changes.